### PR TITLE
Visual studio 2017 support

### PR DESCRIPTION
**Pull request by kovalikp**
Contains two major changes: 
- Update NuGet package to 4.0.0.
- Reworked VS 2017 extension to be self-contained, so it does not require MSI to install,

Had to move some files around, e.g. link .xaml and .cs files from shared project so that rules can be compiled. Changed NuProj.Tasks produces full package Tools output during build, which simplifies NuProj.nuproj considerably.

### CODE REVIEW

**Comment by josesimoes** [comment]
Anything preventing this from being merged and an update published?

**Comment by FranklinTimmy** [comment]
Eagerly waiting for this to be committed, 👍

**Comment by DkSKydancer** [comment]
Eagerly waiting this to be merged!

File: appveyor.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 version: 1.0.{build}
-os: Visual Studio 2015
+os: Visual Studio 2017
```
Comment (AArnott) [inline appveyor.yml]: Is this really the only change you need to appveyor? Is building with MSBuild 14.0 still appropriate?

**Comment by AArnott** [comment]
@terrajobst Why isn't there a build validation result with this PR? Does appveyor not build PRs for this project?

**Comment by kovalikp** [comment]
Sorry I haven't updated this for some time. I've been looking into making VS2017 only build. At the same time I've found that that MSBuild 15 adds another property to `_MSBuildProjectReferenceExistent`. This causes almost all unit tests to fail when building under new version of MSBuild, due to duplicate project builds. Unit tests succeed in MSBuild 14.

**Comment by AArnott** [comment]
That's fine. I have no objection to MSBuild 14 if it works.

**Comment by AArnott** [comment]
And I don't mind if we merge this PR while it still builds support for all versions of VS.

**Comment by bbilginn** [comment]
Eagerly waiting this to be merged!

**Comment by josesimoes** [comment]
Can we please have this merged and published? Not having a decent solution for building Nuget packages in VS 2017 is truly annoying... 😉

**Comment by omegabigdata** [comment]
When does release appear?

**Comment by AArnott** [comment]
It's on nuget.org now. The VSIX isn't on the gallery yet, pending some validation. But you can get it from our CI server's artifact drop, which is linked to from our readme.

**Pull request merged**